### PR TITLE
Feature/new plugins

### PR DESCRIPTION
**Pull request by LiorAgnin**
## Proposed changes

- Bridge to ethereum plugins
- Fuseswap pool plugin

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Refactor or a modification (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: dapp/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "start": "NODE_ENV=development webpack-dev-server --open",
     "dev": "NODE_ENV=development webpack --mode development",
-    "build": "webpack --mode production",
+    "build": "NODE_ENV=production webpack --mode production",
```
Comment (leonprou) [inline dapp/package.json:9]: I think we had a problem with this

Comment (leonprou) [reply dapp/package.json:9]: it's overriding the docker values

File: dapp/src/components/common/withBridge/index.jsx [hunk 2]
```diff
@@ -23,70 +25,136 @@ export default function withBridge(WrappedComponent) {
...
-    const { web3Context } = network
-    const { tokenContext } = dashboard
+    const { tokenContext, _web3Foreign } = dashboard
     const { web3: web3Home } = tokenContext
-    const { web3: web3Foreign } = web3Context
-    const { foreignNetwork, bridgeDirection } = dashboard
+    const { foreignNetwork } = dashboard
     const bridge = dashboard?.bridgeStatus?.from?.bridge
+    const balanceKey = dashboard?.bridgeStatus?.from?.balanceKey
+    const fromBalanceKey = dashboard?.bridgeStatus?.from?.balanceKey
```
Comment (leonprou) [inline dapp/src/components/common/withBridge/index.jsx:35]: balanceKey and fromBalanceKey are the same value

File: dapp/src/components/common/withBridge/index.jsx [hunk 3]
```diff
@@ -23,70 +25,136 @@ export default function withBridge(WrappedComponent) {
...
+      promise.on('receipt', receipt => {
+        setReceipt(receipt)
+        if (Number(receipt.status)) {
+          setTransactionStatus(SUCCESS)
+          onConfirmation()
+        } else {
+          setTransactionStatus(FAILURE)
+        }
+      })
+      // promise.on('receipt', receipt => {
```
Comment (leonprou) [inline dapp/src/components/common/withBridge/index.jsx:127]: can you delete comments?

File: dapp/src/store/mobx/state/dashboard.js [hunk 4]
```diff
@@ -445,10 +457,81 @@ export default class Dashboard {
...
   }
+
+  get bridgeStatus () {
+    if (this?.community?.bridgeDirection === 'home-to-foreign') {
+      if (this.rootStore.network.networkId === 122) {
+        return {
+          from: {
+            network: this.rootStore.network.homeNetwork,
+            bridge: 'foreign',
+            balanceKey: 'fuse'
```
Comment (leonprou) [inline dapp/src/store/mobx/state/dashboard.js:473]: Can we add more data here, like web3 instance and tokenAddress?
 I would love to use less `balanceKey`, and have most of the data given.

File: dapp/src/utils/bridge.js [hunk 5]
```diff
@@ -13,23 +15,45 @@ export const getBridgeMediator = (bridgeType, networkType) => {
   return bridgeMediator
 }
 
-export function approveToken({ networkType, tokenAddress, amount, bridgeType = 'home' }, {
+export const homeToForeignBridgeMediators = {
+  fuse: isProd
```
Comment (leonprou) [inline dapp/src/utils/bridge.js:19]: Can we move that logic to the CONFIG? 
having fields like:
`CONFIG.web3.bridge.addresses.homeToForeign.home` and CONFIG.web3.bridge.addresses.homeToForeign.foreign`

File: server/src/routes/api/v1/communities.js [hunk 6]
```diff
@@ -103,6 +104,30 @@ router.put('/:communityAddress/secondary', async (req, res) => {
...
+ * @apiParam {bridgeType} bridgeType - 'multiple-erc20-to-erc20'/'multi-amb-erc20-to-erc677'/'amb-erc677-to-erc677'
+ * @apiParam {bridgeDirection} bridgeDirection - 'foreign-to-home'/'home-to-foreign'
+ * @apiParamExample {json} Request-Example:
+ *   {
+ *      "foreignTokenAddress": {{foreignTokenAddress}}
+ *   }
+ *
+ */
+
+router.post('/:communityAddress/foreignToken', async (req, res, next) => {
```
Comment (leonprou) [inline server/src/routes/api/v1/communities.js:121]: can you make it PUT, for updating?

File: server/src/routes/api/v1/communities.js [hunk 7]
```diff
@@ -133,6 +158,29 @@ router.post('/:communityAddress/invite', async (req, res, next) => {
...
+ * @apiParam {bridgeDirection} bridgeDirection - 'foreign-to-home'/'home-to-foreign'
+ * @apiParamExample {json} Request-Example:
+ *   {
+ *      "bridgeType": "multi-amb-erc20-to-erc677",
+ *      "bridgeDirection": "home-to-foreign"
+ *   }
+ *
+ */
+
+router.post('/:communityAddress/bridge', async (req, res, next) => {
```
Comment (leonprou) [inline server/src/routes/api/v1/communities.js:176]: PUT too

**Review by leonprou — CHANGES_REQUESTED** [review]
Looks cool 👍 
wrote some comments,  is it possible to make the bridge Dapp part cleaner

File: dapp/src/components/common/withBridge/index.jsx (hunk 2 above)
Comment (LiorAgnin) [reply dapp/src/components/common/withBridge/index.jsx:35]: Thnaks

Comment (LiorAgnin) [reply dapp/src/components/common/withBridge/index.jsx:35]: Thanks

File: dapp/config/default.js [hunk 8]
```diff
@@ -66,6 +68,24 @@
       "network": {
         "homeNetwork": "fuse",
         "foreignNetwork": "ropsten"
+      },
+      "addresses": {
+        "homeToForeign": {
+          fuse: {
+            MultiBridgeMediator: defer(function () {
```
Comment (leonprou) [inline dapp/config/default.js:75]: you can put the prod values on `production.json`

File: server/src/routes/api/v1/communities.js [hunk 9]
```diff
@@ -103,6 +104,30 @@ router.put('/:communityAddress/secondary', async (req, res) => {
   return res.json({ data: community })
 })
 
+/**
+ * @api {post} /communities/:communityAddress/foreignToken Add bridge to community
```
Comment (leonprou) [inline server/src/routes/api/v1/communities.js:108]: please change also here

File: server/src/routes/api/v1/communities.js [hunk 10]
```diff
@@ -133,6 +158,29 @@ router.post('/:communityAddress/invite', async (req, res, next) => {
   }
 })
 
+/**
+ * @api {post} /communities/:communityAddress/bridge Add bridge to community
```
Comment (leonprou) [inline server/src/routes/api/v1/communities.js:162]: and here

File: dapp/src/components/common/withBridge/index.jsx [hunk 11]
```diff
@@ -23,96 +25,135 @@ export default function withBridge(WrappedComponent) {
...
+      fetchHomeNewTokenRegistered({
+        bridgeType,
+        bridgeDirection,
+        fromBlock: blockNumber
+      }, {
+        web3: getWeb3({ networkType: foreignNetwork })
+      }).then((events) => {
+        const newTokenRegisteredRelayEvent = getRelayEventByTransactionHash({ events, transactionHash, accountAddress })
+        if (newTokenRegisteredRelayEvent) {
+          if (get(newTokenRegisteredRelayEvent, 'returnValues.homeToken')) {
```
Comment (leonprou) [inline dapp/src/components/common/withBridge/index.jsx:91]: Can you disable the interval if the is already created? In that case the tx is confirmed but the NewTokenRegistered event is not found

**Review by leonprou — CHANGES_REQUESTED** [review]
Cool changes 👍 
Had another look, I understand this better now. Wrote you some small fixes (but the last one maybe)

File: dapp/config/qa.json [hunk 12]
```diff
@@ -1,7 +1,19 @@
 {
   "env": "qa",
   "web3": {
-    "fuseProvider": "https://rpc.fuse.io"
+    "fuseProvider": "https://rpc.fuse.io",
+    "bridge": {
+      "addresses": {
+        "homeToForeign": {
+          "fuse": {
+            "MultiBridgeMediator": "0x6A90210fc7643Fd4E829204D526C060f0fc347D6"
```
Comment (leonprou) [inline dapp/config/qa.json:9]: @LiorAgnin not needed here, cause when it's not in `qa.json` it takes the default

**Review by leonprou — APPROVED** [review]

**Review by leonprou — APPROVED** [review]

**Pull request merged**